### PR TITLE
feat(persist): persist API-mutated jobs and disable flags across restart

### LINKED ISSUE #593: Persist manually added/edited jobs
Through the Web-UI or API one can manually add or update jobs. However it appears this data is not persisted. That means when restarting the Ophelia container, these jobs are gone / config changes are lost. So it would be nice if these jobs could be persisted somehow so we can preserve our data.

Cheers :)

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Closes #593. Web/UI-created jobs and disable flags previously lived only in scheduler memory, so a `docker compose restart` or pod recycle wiped them. The original reporter ([@FrankyBoy](https://github.com/FrankyBoy)) asked for "some known location we can volume-mount" so the data survives.

## What this adds

- **New `core/persist` package** — JSON Store that atomically writes API state (tmp+rename) on every mutation. On-disk schema carries a `Version` field for explicit forward-migration.
- **New `--state-file` / `OFELIA_STATE_FILE` daemon flag** — empty (default) disables persistence so the pre-#593 behavior is preserved for operators who haven't opted in.
- **Web handler hooks**:
  - create/update: persists API-origin jobs on success
  - delete: now refuses non-API-origin jobs with 403 (was: silently removed any job from memory until next reload — a pre-existing sharp edge)
  - disable/enable: persists the disable flag regardless of origin so a paused INI/label job stays paused after restart
- **Boot order**: INI + Docker labels → state file loads last and shadows same-named INI/label jobs; persisted disable flags apply on top.

## Design decisions locked with the user before code

| Decision | Choice |
|---|---|
| File format | Plain JSON both ways (JSON5 considered; write-side library support for emitting comments is poor) |
| Delete of non-API jobs | Forbid; use `/disable` instead — disable IS persisted, no tombstone construct needed |
| Write-failure handling | Log + return success (best-effort); rolling back would surprise callers |

## Tests

- `core/persist/store_test.go` — 10 round-trip tests including a race catcher that initially failed 82/100 entries surviving, atomic-rename invariant (no .tmp siblings), deterministic-bytes property, missing-file / malformed-JSON / future-version edge cases
- `web/persist_integration_test.go` — 6 end-to-end tests via real `httptest` server + real on-disk file, one per handler hook

## Test plan

- [ ] CI green (full Go test suite + lint)
- [ ] Sample compose flow: `docker compose up`, POST a job via the web UI, `docker compose restart`, verify the job survives
- [ ] Try to delete an INI-defined job via API → expect 403 with origin name
- [ ] Try to disable an INI-defined job via API → expect success; restart → expect still disabled

### CODE REVIEW

**Comment by CybotTM** [comment]
Addressed the multi-axis review (code-reviewer + test-engineer + security-auditor) in 6850ec5.

## Fixed

| Severity | Finding | Fix |
|---|---|---|
| Security MEDIUM | Validation parity gap on boot load | persistedJobToScheduler now runs the same CommandValidator + name checks as the API path |
| Security LOW | File mode implicit via umask | Explicit `Chmod(0o600)` on tempfile + parent dir hardened to `0o700` |
| Security LOW | No size cap on Load → boot OOM | 16 MiB cap with new `ErrStateFileTooLarge` sentinel |
| Security LOW | Persisted origin not re-established at boot → delete-gate bypassed silently | New `MarkOriginAPI(name)` setter, daemon calls it for every persisted job |
| Security LOW (Info) | Typo'd fields silently dropped | `DisallowUnknownFields` on decode |
| Code-review IMPORTANT | Delete-gate change buried in Added | Promoted to "BREAKING (API behavior)" bullet under ### Changed |
| Code-review LOW | "api" literal duplicated 6× | Extracted `originAPI` constant |

## Tests added (test-engineer gaps 7-11)

- `TestStore_Snapshot_IsolatedFromInternalState` — deep-copy contract
- `TestStore_Load_VersionZero_NormalizesToCurrent` — forward-compat
- `TestStore_Load_ZeroByteFile_TreatsAsEmpty` — recoverable empty file
- `TestStore_Load_RejectsUnknownFields` — typo surfaces in error
- `TestStore_Load_RejectsOversizeFile` — 16 MiB cap with sentinel
- `TestStore_Save_FilePermissions` — asserts 0o600 explicitly
- `TestStore_RemoveThenReadd_PersistsNewState` — rename round-trip

## Deferred (with code-comment rationale)

- **code-reviewer Critical "boot-time race with docker label sync"** — real but benign in current scheduler model (gocron's internal locking makes it last-write-wins). A proper fix needs label-sync to be aware of API origins; that's a larger refactor than this PR.
- **code-reviewer Critical "saveLocked holds mutex across disk I/O"** — latency hazard under contended writes (rare for this use case). The separate `saveMu` pattern is doable but adds complexity without evidence of real-world impact yet.
- **code-reviewer "round-trip drops fields if API expands"** — added CONTRACT docstring on `persistedJobToScheduler` so future API field additions get flagged at code review.

All tests + lint green locally.

File: core/persist/store.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+func (s *Store) saveLocked() error {
+	snap := s.snapshotLocked()
+	data, err := json.MarshalIndent(snap, "", "  ")
+	if err != nil {
+		return fmt.Errorf("persist: encode: %w", err)
+	}
+	data = append(data, '\n')
+
+	dir := filepath.Dir(s.path)
+	if err := os.MkdirAll(dir, 0o755); err != nil {
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/persist/store.go]: Added //nolint:gosec G301 with rationale in df5e8dc. 0o700 is intentionally stricter than the 0o750 baseline the rule expects — the state file contains operator-supplied command strings so only the daemon user should have traversal.

File: docs/CONFIGURATION.md [hunk 2]
```diff
@@ -17,6 +17,59 @@ Configuration sources are evaluated in the following order (highest to lowest pr
 2. Environment variables
 3. INI configuration file
 4. Docker labels
+5. **API state file** (`--state-file`) — for jobs created via the web UI or REST API; loads last and shadows same-named INI/label jobs
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply docs/CONFIGURATION.md]: Fixed in df5e8dc. Split the precedence list into two: daemon/global settings (CLI > env > INI > labels, unchanged) and job state (state file > labels > INI, with state file as 'highest').

File: cli/daemon.go [hunk 3]
```diff
@@ -439,3 +473,111 @@ func waitForServerWithErrChan(ctx context.Context, addr string, errChan <-chan e
...
+				"job", name, "error", err)
+			continue
+		}
+		// Replace any same-named job from INI/labels — persisted API
+		// state is authoritative for its own jobs. Use UpdateJob when
+		// the slot exists, else AddJob.
+		if existing := c.scheduler.GetAnyJob(name); existing != nil {
+			_ = c.scheduler.RemoveJob(existing)
+		}
+		if err := c.scheduler.AddJob(job); err != nil {
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply cli/daemon.go:528]: The MarkOriginAPI call in cli/daemon.go already sets the in-memory origins map BEFORE the web server accepts requests (Start happens after boot()), and jobOrigin() reads the in-memory map FIRST before falling back to config reflection (web/server.go:280-285). So jobOrigin returns 'api' for persisted jobs, not 'ini'/'label'. Added clarifying comment + test TestPersist_DeleteJob_RemovesFromStateFileWhenAPIOrigin already covers the happy path.

File: core/persist/store.go [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+		return fmt.Errorf("%w: file has version %d, this build supports up to %d",
+			ErrUnsupportedVersion, loaded.Version, CurrentVersion)
+	}
+	if loaded.Version == 0 {
+		loaded.Version = CurrentVersion
+	}
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.state = loaded
+	return nil
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/persist/store.go:178]: Real bug, fixed in df5e8dc. Load now rejects null entries with a typed ErrNullJobValue sentinel naming the offending key. Added TestStore_Load_RejectsNullJobValue.

File: web/server.go [hunk 5]
```diff
@@ -495,6 +582,16 @@ func (s *Server) createJobHandler(w http.ResponseWriter, r *http.Request) {
...
+	// Persist (#593) only API-origin mutations — INI/labels stay
+	// authoritative for their own jobs. Log + 201 on persist failure
+	// rather than roll back the scheduler insert: the job is already
+	// live in memory and rolling back would surprise callers who
+	// got a successful response shape.
+	if origin == "api" {
+		if err := s.persistJob(req.Name, &req); err != nil {
+			s.scheduler.Logger.Warn("persist created job failed", "job", req.Name, "error", err)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply web/server.go]: Real and worse than reported — pre-fix the entire UI path (X-Origin: web) silently dropped persistence too, because my check was 'if origin == api' which excluded web. Fixed in df5e8dc: client-supplied X-Origin is now coerced back to 'api' if it claims 'ini'/'label', and persistence fires on every successful API mutation regardless of origin. Added TestPersist_CreateJob_IgnoresMaliciousXOriginIni and TestPersist_CreateJob_PersistsWhenOriginWeb.

File: core/persist/store.go [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+	raw, err := os.ReadFile(s.path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		return fmt.Errorf("persist: read %s: %w", s.path, err)
+	}
+	if len(raw) == 0 {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply core/persist/store.go]: Fixed in df5e8dc. Load on a zero-byte file now sets state.Version=CurrentVersion explicitly so the in-memory state matches what the next save would write.

File: cli/daemon.go [hunk 7]
```diff
@@ -439,3 +473,111 @@ func waitForServerWithErrChan(ctx context.Context, addr string, errChan <-chan e
...
+			c.Logger.Warn("skip persisted job (could not materialize)",
+				"job", name, "error", err)
+			continue
+		}
+		// Replace any same-named job from INI/labels — persisted API
+		// state is authoritative for its own jobs. Use UpdateJob when
+		// the slot exists, else AddJob.
+		if existing := c.scheduler.GetAnyJob(name); existing != nil {
+			_ = c.scheduler.RemoveJob(existing)
+		}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply cli/daemon.go:527]: Fixed in df5e8dc. RemoveJob errors during persist apply are now logged with the standard slog Warn pattern naming the job + error.
